patches/tModLoader/Terraria/ModLoader/DamageClass.cs

```diff
@@ -1,31 +1,53 @@
-﻿using Terraria.Localization;
+﻿using Newtonsoft.Json.Serialization;
+using System.Collections.Generic;
+using Terraria.Localization;
 
 namespace Terraria.ModLoader
 {
 	public abstract class DamageClass : ModType
 	{
+		public static Generic Generic => ModContent.GetInstance<Generic>();
+		public static NoScaling NoScaling => ModContent.GetInstance<NoScaling>();
 		public static Melee Melee => ModContent.GetInstance<Melee>();
 		public static Ranged Ranged => ModContent.GetInstance<Ranged>();
 		public static Magic Magic => ModContent.GetInstance<Magic>();
 		public static Summon Summon => ModContent.GetInstance<Summon>();
 		public static Throwing Throwing => ModContent.GetInstance<Throwing>();
 
-		internal int index;
+		/// <summary>
+		/// This is the internal ID of this DamageClass.
+		/// </summary>
+		public int Type { get; internal set; }
 
-		/// <summary> This is the translation that is used behind <see cref="DisplayName"/>. The translation will show up when an item tooltip displays 'X [ClassName]'. This should include the 'damage' part. </summary>
+		/// <summary>
+		/// This is the translation that is used behind <see cref="DisplayName"/>. The translation will show up when an item tooltip displays 'X [ClassName]'. This should include the 'damage' part.
+		/// </summary>
 		public ModTranslation ClassName { get; internal set; }
 
 		/// <summary> This is the name that will show up when an item tooltip displays 'X [ClassName]'. This should include the 'damage' part. </summary>
 		public string DisplayName => DisplayNameInternal;
+		/// <summary>
+		/// This lets you define the classes that this DamageClass will benefit from (other than itself) for the purposes of stat bonuses, such as damage and crit chance.
+		/// Returns 0 in all cases by default, which does not let any other classes boost this DamageClass.
+		/// </summary>
+		/// <param name="damageClass">The DamageClass which you want this DamageClass to benefit from statistically.</param>
+		public virtual float BenefitsFrom(DamageClass damageClass) => 0;
```

Comment:
Yep, on it right now. Rather than internal though, I made it private and added a public method that will handling caching/retrieving the value. It should both be cleaner on the backend but also allow for modders to safely access it if need be.

For now, I'm naming this new method `GetCachedBenifitsFrom`, we can change it latter if needed though.